terragrunt/aws/iam.tf

```diff
@@ -0,0 +1,80 @@
+data "aws_iam_policy_document" "service_principal" {
+  statement {
+    effect = "Allow"
+
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "web" {
+  name               = "${var.product_name}-web"
+  assume_role_policy = data.aws_iam_policy_document.service_principal.json
+}
+
+data "aws_iam_policy" "lambda_insights" {
+  name = "CloudWatchLambdaInsightsExecutionRolePolicy"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_insights" {
+  role       = aws_iam_role.web.name
+  policy_arn = data.aws_iam_policy.lambda_insights.arn
+}
+
+data "aws_iam_policy_document" "web_policies" {
+
+  statement {
+
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+
+    resources = [
+      "arn:aws:logs:${var.region}:${var.account_id}:log-group:*"
+    ]
+  }
+
+  statement {
+
+    effect = "Allow"
+
+    actions = [
+      "ecr:GetDownloadUrlForlayer",
+      "ecr:BatchGetImage"
+    ]
+    resources = [
+      aws_ecr_repository.web.arn
+    ]
+  }
+
+}
+
+resource "aws_iam_policy" "web" {
+  name   = "${var.product_name}-web"
+  path   = "/"
+  policy = data.aws_iam_policy_document.web_policies.json
+}
+
+resource "aws_iam_role_policy_attachment" "web" {
+  role       = aws_iam_role.web.name
+  policy_arn = aws_iam_policy.web.arn
+}
+
+# Use AWS managed IAM policy
+####
+# Provides minimum permissions for a Lambda function to execute while
+# accessing a resource within a VPC - create, describe, delete network
+# interfaces and write permissions to CloudWatch Logs.
+####
+resource "aws_iam_role_policy_attachment" "AWSLambdaVPCAccessExecutionRole" {
+  role       = aws_iam_role.web.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole"
+}
```

Comment:
You can get rid of this policy attachment since your function isn't in a VPC.  This is needed for things like describing security groups, subnets, etc.